src/webgpu/api/validation/encoding/cmds/render/indirect_draw.spec.ts

```diff
@@ -45,22 +30,28 @@ Tests indirect buffer must be valid.
   .paramsSubcasesOnly(kIndirectDrawTestParams.combine('state', kBufferStates))
   .fn(t => {
     const { encoderType, indexed, state } = t.params;
+    const pipeline = t.createNoOpRenderPipeline();
     const indirectBuffer = t.createBufferWithState(state, {
       size: 256,
       usage: GPUBufferUsage.INDIRECT,
     });
 
     const { encoder, finish } = t.createEncoder(encoderType);
-    encoder.setPipeline(t.pipeline);
+    encoder.setPipeline(pipeline);
     if (indexed) {
-      encoder.setIndexBuffer(t.indexBuffer, 'uint32');
+      const indexBuffer = t.makeIndexBuffer();
+      encoder.setIndexBuffer(indexBuffer, 'uint32');
       encoder.drawIndexedIndirect(indirectBuffer, 0);
     } else {
       encoder.drawIndirect(indirectBuffer, 0);
     }
 
     t.expectValidationError(() => {
-      t.queue.submit([finish()]);
+      if (state === 'destroyed') {
```

Comment:
In this case the finish() call needs to be called outside of expectValidationError. That way if there is erroneously an error from finish(), the test will fail.

However I am planning to do a refactor to make this less painful soon anyway, and this is a very small thing, so I'll update it then.